feat(ai): Updated skill and template for migration plan (#6195)
adobe/spectrum-web-components: .ai/skills/migration-prep/assets/migration-prep-template.md

### Additive — ships when ready, zero breakage for consumers already on 2nd-gen

<!--
Summarize additive aspects across the following categories.

# Format: **A#**
-->
<!-- These items are usually deferred or out of scope for the current migration unless explicitly included. They are additive/deferred, not deprecated or removed. -->

| #   | What is added | Notes |
| --- | ------------- | ----- |

---

## 2nd-gen API decisions

These are derived from the 1st-gen implementation, current deprecations, the Figma Desktop [component] spec, the React S2 implementation, and the rendering roadmap. Confirmed items are marked; open items are tracked in [Blockers and open questions](#blockers-and-open-questions).

Use lightweight confidence labels where helpful:

- **Confirmed**: directly supported by source material
- **Inferred**: recommended based on multiple signals, but not explicitly specified in one authoritative source
- **Open question**: unresolved and needs review or more input

### Public API

#### Properties / attributes (2nd-gen)

| Property | Type | Default | Attribute | Notes |
| -------- | ---- | ------- | --------- | ----- |

#### Visual matrix (2nd-gen)

<!--

> Example content for this section - may be ommitted if not relevant.

Reviewer: The skill calls out `Prefer marking truly inapplicable sections as `N/A` instead of deleting them unless the user explicitly wants sections removed`. Should we align this section or align the skill with this?

```suggestion
If not applicable, mark as N/A with a brief reason rather than removing the section.
```

### Behavioral semantics

<!-- Include as many sections as are necessary to address the component's unique aspects. Callout any non-obvious implementation details. -->

### Accessibility semantics notes (2nd-gen)

---

## Architecture: core vs SWC split

> The 1st-gen component is a **reference only** — 2nd-gen is built independently. Neither generation imports from the other.

Follow the [Badge migration reference](../../02_workstreams/02_2nd-gen-component-migration/02_step-by-step/01_washing-machine-workflow.md#reference-badge-migration) as the concrete pattern for the core/SWC split.

<!-- Use the prescribed table format below; do not convert this section to bullets or another schema. -->

| Layer    | Path                                            | Contains                                                                                                                                                                                |
| -------- | ----------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | **Core** | `2nd-gen/packages/core/components/[component]/` | `[Component].base.ts`, `[Component].types.ts`, validation, state, accessible-name logic, attribute forwarding, pending-label behavior, and other reusable semantic rules. No rendering. |

Reviewer 2: ```suggestion
| **Core** | `2nd-gen/packages/core/components/[component]/` | `[Component].base.ts`, `[Component].types.ts`, validation, state, accessible-name logic, attribute forwarding, and other reusable semantic rules. No rendering. Add component-specific reusable behaviors here as needed. |
```

- [ ] Add `.swc-[Component]` to the internal semantic `<[component]>` in `render()`; keep styling off `:host`
- [ ] Copy S2 source from `spectrum-css` `spectrum-two` branch `index.css` (not `/dist`) into `[component].css` as baseline

#### Visual model and regressions

<!-- Most components will need these items; add others as needed -->

- [ ] Verify i18n size modifiers (`:lang(ja)`, `:lang(ko)`, `:lang(zh)`) if present in S2 source
- [ ] Pass stylelint (property order, `no-descending-specificity`, token validation)

### Accessibility

<!-- Accessibility checklist items should be sourced from the `accessibility-migration-analysis.md` which is a pre-requisite to this migration plan. If it is not available, include a checklist item/open question requiring it before this plan is considered complete. -->

#### Naming and semantics

#### State verification

### Testing

<!-- Fill in comprehensive test cases -->

- [ ] Port `1st-gen/packages/[component]/test/[component].test.ts` coverage that still applies
- [ ] Add Playwright `[component].a11y.spec.ts` with `toMatchAriaSnapshot`

#### Behavior

#### Visual regression

<!--

> Retain this section for any components with visual rendering, modifying as needed for the component's specs and variants.

Reviewer 2: ```suggestion
Retain this section for any components with visual rendering, modifying as needed for the component's specs and variants. Replace the example bullets below with VRT items that match this component, and reference real bug tickets only when they apply to this component.

Example items (replace with this component's variants and any related bug tickets — do not carry these examples forward verbatim):
```

- [ ] Add VRT coverage for size, variant, treatment, static color, and pending combinations
- [ ] Add visual regression coverage for static white outline on its approved background, including hover state (`SWC-1139`)
- [ ] Add visual/high-contrast coverage for the pending state using disabled styling (`SWC-459`)

> - [ ] Add focus-visible regression coverage for truncated buttons so the ring is not clipped (`SWC-886`) -->

Reviewer 2: ```suggestion
- [ ] Add VRT coverage for the component's size, variant, treatment, and state combinations
- [ ] Add visual regression coverage for any static color treatments on their approved backgrounds, including hover state (`SWC-####`)
- [ ] Add visual/high-contrast coverage for any state that relies on shared styling (e.g. pending using disabled styling) (`SWC-####`)
- [ ] Add focus-visible regression coverage where the focus ring may be clipped or obscured (`SWC-####`) -->
```

-->

<!-- Keep the table structure and columns (`Blocking?`, `Status`, `Owner`). Add rows rather than replacing the schema with prose or a different format. -->

### Design

| #   | Item | Blocking? | Status | Owner |
| --- | ---- | --------- | ------ | ----- |

### Architecture and behavior

| #   | Item | Blocking? | Status | Owner |
| --- | ---- | --------- | ------ | ----- |

### Scope and prerequisites

| #   | Item | Blocking? | Status | Owner |
| --- | ---- | --------- | ------ | ----- |

<!-- Where possible, include the next action in the Item text or Status so reviewers know how to resolve the question. -->

---

## References

<!-- IMPORTANT: Use as references for compiling data and details and informing decisions for all other sections of this plan. -->
<!-- Do not leave this section sparse. Include the analysis docs, 1st-gen implementation references, relevant design references, and any bug tickets that informed decisions. -->

- [Washing machine workflow](../../02_workstreams/02_2nd-gen-component-migration/02_step-by-step/01_washing-machine-workflow.md)
- [2nd-gen migration status table](../../02_workstreams/02_2nd-gen-component-migration/01_status.md)

> - [Accessibility migration analysis](./accessibility-migration-analysis.md)

Reviewer 3: I tested this without the a11y migration analysis, it might be helpful to note that if one is not present, this link needs to be adjusted/removed or we need to note that there isn't one? (It might already do this last part, I didn't end up with a broken link but rather something more like this, which is ok.)

```
- **Accessibility migration analysis** — add as `./accessibility-migration-analysis.md` when the accessibility workstream produces it (no link until the file exists; avoids broken doc links).
```

Author: added, but following the pattern based on Rajdeep's suggestions for these links

- [Rendering and styling migration analysis](./rendering-and-styling-migration-analysis.md)
- [CSS style guide — Component Custom Property Exposure](../../../../CONTRIBUTOR-DOCS/02_style-guide/01_css/02_custom-properties.md#component-custom-property-exposure)
- [CSS style guide — Selector conventions](../../../../CONTRIBUTOR-DOCS/02_style-guide/01_css/02_custom-properties.md#selector-conventions)
- [1st-gen source](../../../../1st-gen/packages/[component]/src/[Component].ts)
- [1st-gen shared base / mixins](../../../../1st-gen/packages/[component]/src/)
- [1st-gen tests](../../../../1st-gen/packages/[component]/test/[component].test.ts)
- [1st-gen README](../../../../1st-gen/packages/[component]/README.md)

> - [React Spectrum S2 [Component]](https://react-spectrum.adobe.com/[Component])

Reviewer 2: ```suggestion
- [1st-gen shared base / mixins](../../../../1st-gen/packages/[component]/src/) — include only if the component has shared base or mixin files
- [1st-gen tests](../../../../1st-gen/packages/[component]/test/[component].test.ts) — include only if a test file exists
- [1st-gen README](../../../../1st-gen/packages/[component]/README.md) — include only if a README exists
- [React Spectrum S2 [Component]](https://react-spectrum.adobe.com/[Component]) — confirm the URL slug matches the actual React Spectrum docs page
```

- [Rendering and styling migration analysis](./rendering-and-styling-migration-analysis.md)
- [CSS style guide — Component Custom Property Exposure](../../../../CONTRIBUTOR-DOCS/02_style-guide/01_css/02_custom-properties.md#component-custom-property-exposure)
- [CSS style guide — Selector conventions](../../../../CONTRIBUTOR-DOCS/02_style-guide/01_css/02_custom-properties.md#selector-conventions)
- [1st-gen source](../../../../1st-gen/packages/[component]/src/[Component].ts)
- [1st-gen shared base / mixins](../../../../1st-gen/packages/[component]/src/) — include only if the component has shared base or mixin files
- [1st-gen tests](../../../../1st-gen/packages/[component]/test/[component].test.ts) — include only if a test file exists
- [1st-gen README](../../../../1st-gen/packages/[component]/README.md) — include only if a README exists
- [React Spectrum S2 [Component]](https://react-spectrum.adobe.com/[Component]) — confirm the URL slug matches the actual React Spectrum docs page
- [Badge migration reference](../../02_workstreams/02_2nd-gen-component-migration/02_step-by-step/01_washing-machine-workflow.md#reference-badge-migration)
- Epic: SWC-#### - [Component] epic

> 

Reviewer 3: Optionally, would it be helpful to also include Spectrum-CSS source (probably further up than where I'm suggesting)? The migration prep I just ran tried to add it, but pointed to the main branch.

```suggestion
- [Spectrum CSS — `spectrum-two` branch](https://github.com/adobe/spectrum-css/tree/spectrum-two) — S2 styling source of truth for the component; baseline lives under `components/[component]/` (typically `index.css`, not `/dist`). Keep a sibling checkout of `spectrum-css` on **`spectrum-two`** while migrating (see [Setup](#setup) checklist).
```

Author: oh yes, really good call - added!
